mlir/lib/Conversion/MathToLLVM/MathToLLVM.cpp

```diff
@@ -286,6 +286,40 @@ struct RsqrtOpLowering : public ConvertOpToLLVMPattern<math::RsqrtOp> {
   }
 };
 
+struct IsNaNOpLowering : public ConvertOpToLLVMPattern<math::IsNaNOp> {
+  using ConvertOpToLLVMPattern<math::IsNaNOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(math::IsNaNOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto operandType = adaptor.getOperand().getType();
+
+    if (!operandType || !LLVM::isCompatibleType(operandType))
+      return failure();
+
+    rewriter.replaceOpWithNewOp<LLVM::IsFPClass>(op, op.getType(),
+                                                 adaptor.getOperand(), 3);
```

Comment:
https://github.com/llvm/llvm-project/blob/e729dc759d052de122c8a918fe51b05ac796bb50/llvm/include/llvm/ADT/FloatingPointMode.h#L239

Can we use the constants here?